### PR TITLE
Cleanup Sphinx conf

### PR DESCRIPTION
**Pull request by saimn**
Cleanup a bit the config file while looking at sphinx_astropy's defaults.

### Description

This pull request is to address ...

Fixes #<Issue Number>

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [ ] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [ ] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by nstarman** [comment]
@saimn , is this ready for review?

**Comment by saimn** [comment]
@nstarman - Yes, just waiting for a kind soul to review :)

File: docs/conf.py [hunk 1]
```diff
@@ -85,54 +88,51 @@
 # -- General configuration ----------------------------------------------------
 
 # If your documentation needs a minimal Sphinx version, state it here.
-needs_sphinx = "1.7"
-
-# To perform a Sphinx version check that needs to be more specific than
-# major.minor, call `check_sphinx_version("X.Y.Z")` here.
-check_sphinx_version("1.2.1")  # noqa: F405
+needs_sphinx = "3.0"
```
Comment (nstarman) [inline docs/conf.py:91]: This is a big bump, but appears to be fine. What version of sphinx are we actually building with?

File: docs/conf.py [hunk 2]
```diff
@@ -376,11 +338,6 @@
 linkcheck_timeout = 180
 linkcheck_anchors = False
 
-# Add any extra paths that contain custom files (such as robots.txt or
-# .htaccess) here, relative to this directory. These files are copied
-# directly to the root of the documentation.
-html_extra_path = ["robots.txt"]
```
Comment (nstarman) [inline docs/conf.py:382]: Perhaps we should instead add robots.txt. Or open an issue to add one!

File: docs/conf.py (hunk 1 above)
Comment (saimn) [reply docs/conf.py:91]: 3.0 is from April 2020 :
https://pypi.org/project/Sphinx/#history

File: docs/conf.py (hunk 2 above)
Comment (saimn) [reply docs/conf.py:382]: This was moved above in the HTML section.

**Review by nstarman — APPROVED** [review]
My questions are resolved. Thanks @saimn for the simplifications!

**Pull request merged**